keyboards/sporewoh/banime40/config.h

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2022 sporewoh
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xBEAF
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0000
+#define MANUFACTURER    sporewoh
+#define PRODUCT         banime40
+
+/* key matrix size */
+#define MATRIX_ROWS 4
+#define MATRIX_COLS 10
+
+/* pin-out for PCB */
+#define MATRIX_ROW_PINS { E6, D7, C6, D4 }
+#define MATRIX_COL_PINS { B4, B5, B6, B2, B3, B1, F7, F6, F5, F4 }
+
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 10
```

Comment:
If this is set to 10, then you need to have 10 layers defined for the VIA keymap.  This way that all of the layers are populated proper on initialization, rather than filled with random junk. 

That or, this needs to be set to 4 (or removed, as 4 is the default)

Reply:
Good catch, I've added the 6 missing layers (I might need more than 4 layers for a keyboard of this size 😳)

Reply:
This board appears to be using an atmega32u4, which, at best is pushing the 1kb of eeprom space that the controller has.

Does the VIA keymap even compile?

Reply:
Yes, the VIA keymap compiles and is fully usable in VIAL. I'm guessing one of the reasons why I can get away with it, despite using an ATmega32U4, is just due to the key count of the board